File: src/HotelOrderSystem.Api/Dtos/PresenceDtos.cs
using System.ComponentModel.DataAnnotations;

namespace HotelOrderSystem.Api.Dtos;

public sealed record HeartbeatRequest(
    [property: Required, MaxLength(200)] string DeviceId,
    [property: MaxLength(50)] string AppState,
    [property: MaxLength(100)] string? CurrentScreen);

public sealed record HeartbeatResponse(
    DateTime ServerTimeUtc,
    bool Online,
    bool IsReady,
    int PendingOrdersCount);

public sealed record AvailabilityRequest(
    bool IsReady,
    string? DeviceId,
    string? Source);

public sealed record AvailabilityResponse(
    bool IsReady,
    DateTime? ReadySinceAtUtc,
    DateTime ChangedAtUtc);

public sealed record StaffPresenceDto(
    int UserId,
    string FullName,
    int? TeamId,
    string? TeamName,
    bool IsOnline,
    bool IsReady,
    DateTime? ReadySinceAtUtc,
    DateTime? LastHeartbeatAtUtc,
    string? LastKnownAppState);

File: android-staff-app/app/src/main/java/com/ibaapps/HotelOrderSystem/data/remote/dto/PresenceDtos.kt
package com.ibaapps.HotelOrderSystem.data.remote.dto

import kotlinx.serialization.Serializable

@Serializable
data class HeartbeatRequestDto(
    val deviceId: String,
    val appState: String,
    val currentScreen: String? = null
)

@Serializable
data class HeartbeatResponseDto(
    val serverTimeUtc: String,
    val online: Boolean,
    val isReady: Boolean,
    val pendingOrdersCount: Int
)

@Serializable
data class AvailabilityRequestDto(
    val isReady: Boolean,
    val deviceId: String? = null,
    val source: String? = null
)

@Serializable
data class AvailabilityResponseDto(
    val isReady: Boolean,
    val readySinceAtUtc: String? = null,
    val changedAtUtc: String
)
